include/oneapi/tbb/concurrent_queue.h

```diff
@@ -178,8 +182,12 @@ class concurrent_queue {
     }
 
     void swap ( concurrent_queue& other ) {
-        //TODO: implement support for std::allocator_traits::propagate_on_container_swap
-        __TBB_ASSERT(my_allocator == other.my_allocator, "unequal allocators");
+        if (queue_allocator_traits::propagate_on_container_swap::value) {
+            using std::swap;
+            swap(my_allocator, other.my_allocator);
+        } else {
+            __TBB_ASSERT(my_allocator == other.my_allocator, "unequal allocators");
```

Comment:
Lets move it inside `internal_swap`

Reply:
I reviewed all the containers’ `internal_swap`, which typically have two overloads. I think they can be combined into a single function using `if`:

```cpp
void internal_swap(concurrent_queue& src) {
   if (!allocator_traits_type::queue_allocator_traits::propagate_on_container)
        __TBB_ASSERT(my_allocator == src.my_allocator, "Swapping with unequal allocators is not allowed");
    using std::swap;
    swap(my_queue_representation, src.my_queue_representation);
}
```
The compiler will optimize it and eliminate the unreachable branch.
Allocator helpers have the same pattern. Do we need to perform these cleanups, or should we keep things as they are?